### ISSUE TITLE
Using metadata queries in python

### ISSUE THREAD

**Issue by jjkoehorst** [irods/python-irodsclient#158]
I am trying to query metadata fields from within python such that the following query can be used.

> imeta qu -C type like Project

What would be the correct approach as I could not really find any documentation about this or was I searching on the wrong location?

**Comment by trel** [comment]
You can use the General Query interface and some filtering to achieve the same results.

Prepare the test:
```
$ imeta add -C /tempZone/home/rods type Project

$ imeta ls -C /tempZone/home/rods
AVUs defined for collection /tempZone/home/rods:
attribute: type
value: Project
units: 

$ imeta qu -C type like Project
collection: /tempZone/home/rods
```

The test code `158.py`:
```python
from irods.column import Criterion
from irods.models import DataObject, DataObjectMeta, Collection, CollectionMeta
from irods.session import iRODSSession
import os
env_file = os.path.expanduser('~/.irods/irods_environment.json')
with iRODSSession(irods_env_file=env_file) as session:
    # by metadata
    # equivalent to 'imeta qu -C type like Project'
    print('----')
    results = session.query(Collection, CollectionMeta).filter( \
        Criterion('=', CollectionMeta.name, 'type')).filter( \
        Criterion('like', CollectionMeta.value, '%Project%'))
    for r in results:
        print(r[Collection.name], r[CollectionMeta.name], r[CollectionMeta.value], r[CollectionMeta.units])
```

Run it:
```
$ python 158.py 
----
('/tempZone/home/rods', 'type', 'Project', None)
```

**Comment by trel** [comment]
I've updated the README.rst in 800488b7b3af97342f229945a8da69c931f09180

Please reopen this issue if necessary.

**Comment by jjkoehorst** [comment]
It does indeed work like a charm, however I have a small question. Is it possible to filter for multiple CollectionMeta name and value combinations?

For example, date, and hour? or name and something else?

I think by default it tries to match it all on the same key/value pair resulting in nothing found.

**Comment by trel** [comment]
@jjkoehorst Just looking through unresponded-to messages - we've added some more to the README since last March - please look there to see if your use cases are covered.  If not, please open a new issue.

**Comment by d-w-moore** [comment]
@jjkoehorst  Indeed you can search on multiple name / value / unit combinations, for any type of AVU (CollectionMeta included) . Note that the default when chaining through multiple .filter( ) stages -- or multiple comma separated criteria in the same filter method call -- is to narrow the scope of the search a.k.a. AND/intersection  semantics.  One might think that obvious, ... however ....
There is a caveat here (w.r.t. the GenQuery machinery within iRODS) in that you can also search for multiple AVU values having the same attribute name (resulting an "OR" or union type of search for this special case) by combining consecutive attribute-NAME-comparison criteria via "and".  This is expressed in iquest and LIGQ as eg.:
```
  select COLL_NAME where 
   META_COLL_ATTR_NAME = 'a' and META_COLL_ATTR_VALUE = '1' and 
   META_COLL_ATTR_NAME = 'a' and META_COLL_ATTR_VALUE = '2' 
```
or in Python iRods Client as eg.:
```
session.query(Collection).filter(
  CollectionMeta.name == 'a' and CollectionMeta.value == '1' and
  CollectionMeta.name == 'a' and CollectionMeta.value == '2' )
```
Note for the PRC version of this to be available , one would need 0.8.3, or just pull and `pip install -e .` while on the tip of [master](http://github.com/irods/python-irodsclient).

In each case, the second "and" operator executes much like an "OR".  Units don't follow this special-case semantic however, so be warned.  But for Name/Value searches this is quite useful because otherwise the PRC doesn't really have an OR operator that executes on the SQL level, and the "||" operator in genquery only operates on one field at at a time.
